src/compositions/useDeployments.ts

```diff
@@ -1,55 +1,45 @@
-import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
-import { MaybeRefOrGetter, computed, toRef, toValue } from 'vue'
+import { MaybeRefOrGetter, toValue } from 'vue'
 import { useCan } from '@/compositions/useCan'
+import { PaginationOptions, UsePaginationEntity, usePagination } from '@/compositions/usePagination'
 import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
 import { DeploymentsFilter } from '@/models'
 import { WorkspaceDeploymentsApi } from '@/services'
 import { Getter } from '@/types/reactivity'
-import { UseEntitySubscription } from '@/types/useEntitySubscription'
-import { isNullish } from '@/utilities'
 
-export type UseDeployments = UseEntitySubscription<WorkspaceDeploymentsApi['getDeployments'], 'deployments'>
+export type UseDeployments = UsePaginationEntity<
+WorkspaceDeploymentsApi['getDeployments'],
+WorkspaceDeploymentsApi['getDeploymentsCount'],
+'deployments'
+>
 
-export function useDeployments(filter: MaybeRefOrGetter<DeploymentsFilter>): UseDeployments
-export function useDeployments(deploymentIds: MaybeRefOrGetter<string[] | null | undefined>): UseDeployments
-export function useDeployments(filterOrDeploymentIds?: MaybeRefOrGetter<string[] | DeploymentsFilter | null | undefined>): UseDeployments {
```

Comment:
Decided to remove the overload here since we weren't using it anywhere. Its also simple enough to use a getter for this at implementation like
```typescript
useDeployments(() => ({
 deployments: {
    id: ['one', 'two']
  }
})